pilot/pkg/networking/core/v1alpha3/listener.go

```diff
@@ -660,9 +684,8 @@ func buildListener(opts buildListenerOpts) *xdsapi.Listener {
 // we should encapsulate them some way to ensure they remain consistent (mainly that in each an index refers to the same
 // chain)
 func marshalFilters(l *xdsapi.Listener, opts buildListenerOpts, chains []plugin.FilterChain) error {
-	if len(opts.filterChainOpts) != len(chains) || len(chains) != len(l.FilterChains) || len(opts.filterChainOpts) == 0 {
-		return fmt.Errorf("must have more than 0 chains, and the same number of chains in each of:\nlistener: %d; %#v\nopts: %d; %#v\nchain: %d; %#v",
-			len(l.FilterChains), l, len(opts.filterChainOpts), opts, len(chains), chains)
+	if len(opts.filterChainOpts) == 0 {
```

Comment:
You need the other checks too: you need the same number of chains in each of the fields since `chains[i]` is associated with `opts.filterChainOpts[i]` which is associated with `l.FilterChains[i]`.

Reply:
But Envoy no longer has that restriction. Can have chains of different length.

Reply:
once that PR merges. You think we should still have this condition until then? I am also not sure if this has repercussions for the proto to struct conversion stuff that goes on, as we assume that stuff is laid out in an array in a particular form. cc @ostromart thoughts?

Reply:
My 2c: I don't think it's a big deal to skip the check for now if the change is imminent. But at least I'd leave a TODO here to generally beef up the checks (including this one) when we get around to doing a clean up. It will take some thinking to get the checks right so we aren't spraying them around everywhere and duplicating.

Reply:
+1 on skipping the test.
With valid config it'll work with the old proxy as well, and invalid configs have tons of other problems in how they are handled ( i.e dropping too much)

Reply:
Glad to hear that restriction is gone, dumping the check is fine in that case.